lib/rmt/file_validator.rb

```diff
@@ -0,0 +1,59 @@
+require 'fileutils'
+require 'rmt/checksum_verifier'
+
+class RMT::FileValidator
+  class << self
+    def validate_local_file(file_reference, deep_verify:)
+      verify_file_on_disk(file_reference, deep_verify).tap do |valid_on_disk|
+        update_file_status_on_database(file_reference, valid_on_disk)
+      end
+    end
+
+    def find_valid_file_by_checksum(checksum, checksum_type, deep_verify:)
+      ::DownloadedFile
+        .where(checksum: checksum, checksum_type: checksum_type)
+        .map { |file| [file, validate_tracked_file_on_disk(file, deep_verify)] }
+        .find { |_, valid_on_disk| valid_on_disk }
+        &.first&.local_path
+    end
+
+    private
+
+    def validate_tracked_file_on_disk(file_record, deep_verify)
+      verify_file_on_disk(file_record, deep_verify).tap do |valid_on_disk|
+        file_record.destroy unless valid_on_disk
+      end
+    end
+
+    def verify_file_on_disk(file, deep_verify)
+      file_exist = File.exist?(file.local_path)
```

Comment:
So if the file doesn't exist, `match_metadata = false` and you aren't not removing the file and just returning false.

You can just return false immediately: `return false unless File.exist?(file.local_path)`?

Reply:
Done :heavy_check_mark: